longevity_tombstone_gc_test.py

```diff
@@ -0,0 +1,127 @@
+import datetime
+import re
+import time
+from functools import partial
+
+from longevity_twcs_test import TWCSLongevityTest
+from sdcm.utils.common import ParallelObject
+from sdcm.utils.sstable_utils import SstableUtils
+
+
+class TombstoneGcLongevityTest(TWCSLongevityTest):
+    keyspace = 'scylla_bench'
+    table = 'test'
+    ks_cf = f'{keyspace}.{table}'
+    repair_date = None
+    db_node = None
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        post_prepare_cql_cmds = self.params.get('post_prepare_cql_cmds')
+        self.ttl = int(re.search(r'default_time_to_live = (\d+)', post_prepare_cql_cmds).group(1))
+        self.propagation_delay = int(
+            re.search(r"'propagation_delay_in_seconds':'(\d+)", post_prepare_cql_cmds).group(1))
+
+    def _run_repair_and_major_compaction(self, wait_propagation_delay: bool = False):
+        self.log.info('Run a flush for %s on nodes', self.keyspace)
+        triggers = [partial(node.run_nodetool, sub_cmd=f"flush -- {self.keyspace}", ) for node in self.db_cluster.nodes]
+        ParallelObject(objects=triggers, timeout=1200).call_objects()
+
+        self.log.info('Run a repair for %s on nodes', self.ks_cf)
+        triggers = [partial(node.run_nodetool, sub_cmd="repair", args=f"-pr {self.keyspace} {self.table}", ) for node
+                    in self.db_cluster.nodes]
+        ParallelObject(objects=triggers, timeout=1200).call_objects()
+
+        if wait_propagation_delay:
+            time.sleep(self.propagation_delay)
+        self.repair_date = datetime.datetime.now()
+        self.log.info('Run a major compaction for %s on node', self.ks_cf)
+        self.db_node.run_nodetool("compact", args=f"{self.keyspace} {self.table}")
+        self.wait_no_compactions_running()
+
+    def test_switch_tombstone_gc_modes(self):
```

Comment:
please add purpose of this test before test steps so reviewers can understand and verify steps look reasonable.
Maybe some reference to feature documentation.

Reply:
added.